zrml/neo-swaps/src/lib.rs

```diff
@@ -626,15 +869,28 @@ mod pallet {
                     amount_in <= pool.calculate_numerical_threshold(),
                     Error::<T>::NumericalLimits(NumericalLimitsError::MaxAmountExceeded),
                 );
+
+                let buy = vec![asset_in];
+                let keep = vec![];
+                let sell = pool.assets_complement(&buy);
+                let amount_out = pool.calculate_swap_amount_out_for_sell(
+                    buy,
+                    keep,
+                    sell,
+                    amount_in,
+                    Zero::zero(),
+                )?;
```

Comment:
It's fine, because `asset_in` is sold and in order to get the amount of the full set of all possible outcomes, the `amount_out` is calculated in which all other assets are sold to get an equal amount of each possible asset back, which can then be burned for collateral.